core/src/apps/cardano/sign_tx.py

```diff
@@ -112,19 +148,22 @@ def _build_inputs(inputs: List[CardanoTxInputType]) -> List[Tuple[bytes, int]]:
 
 
 def _build_outputs(
-    keychain: seed.Keychain, outputs: List[CardanoTxOutputType], protocol_magic: int
+    keychain: seed.Keychain,
+    outputs: List[CardanoTxOutputType],
+    protocol_magic: int,
+    network_id: int,
 ) -> List[Tuple[bytes, int]]:
     result = []
     for output in outputs:
         amount = output.amount
-        if output.address_n:
-            address, _ = derive_address_and_node(
-                keychain, output.address_n, protocol_magic
+        if output.address_parameters:
+            address = derive_address_bytes(
+                keychain, output.address_parameters, protocol_magic, network_id
             )
         else:
-            address = output.address
+            address = get_address_bytes_unsafe(output.address)
```

Comment:
please add a comment that the validity of the output address is checked in `_validate_outputs` before this happens

Reply:
Added [here](https://github.com/trezor/trezor-firmware/pull/1112/commits/30e49c48d085fe9ab528f7f5be7ac68421389eec).